### PR TITLE
fix(core/xref): Improve error messages

### PR DESCRIPTION
**Pull request by sidvishnoi**
- Updates error messages

One present issue that needs to be resolved is that the multiple warnings are emitted for same term. It also tells that our `disambiguate` algorithm is executing for each term (even when they are same). So, that fix is coming up next.

image

### CODE REVIEW

**Comment by sidvishnoi** [comment]
Update: 
image

File: src/core/utils.js [hunk 1]
```diff
@@ -293,11 +293,20 @@ export function removeReSpec(doc) {
   });
 }
 
-export function showInlineError(elem, msg) {
-  elem.classList.add("respec-offending-element");
-  elem.setAttribute("title", msg);
-  pub("warn", msg + " See develper console for details.");
-  console.warn(msg, elem);
+/**
+ * Adds error class of each element while emitting a warning
```
Comment (sidvishnoi) [inline src/core/utils.js]: noticed nit 
s/of/to

File: src/core/xref.js [hunk 2]
```diff
@@ -19,12 +19,12 @@ export async function run(conf, elems) {
   const { xref } = conf;
   const xrefMap = createXrefMap(elems);
   const query = createXrefQuery(xrefMap);
-  const apiURL = new URL(xref.url, location.href) || API_URL;
+  const apiURL = (xref.url && new URL(xref.url, location.href)) || API_URL;
```
Comment (marcoscaceres) [inline src/core/xref.js]: Here you want:

```JS
const apiURL = xref.url ? new URL(xref.url, location.href).href : API_URL;
```

Comment (sidvishnoi) [reply src/core/xref.js]: `.href`? need some consistency here
so, which one should change? API_URL or new URL here?

**Pull request merged**